### PR TITLE
add header for http healthcheck

### PR DESCRIPTION
**Pull request by KaiveYoung**

### CODE REVIEW

File: client/health/health.go [hunk 1]
```diff
@@ -163,6 +171,8 @@ func (monitor *Monitor) doHTTPCheck(ctx context.Context) error {
 	if err != nil {
 		return err
 	}
+	req.Header = monitor.header
+	req.Host = monitor.host
```
Comment (fatedier) [inline client/health/health.go:175]: I have some misunderstandings about the configuration of the host in K8s. We should not add the host parameter here, but instead actively extract the host field from the header and assign it to req.Host.

```go
 req.Host = monitor.header.Get("Host")
```

**Review by fatedier — CHANGES_REQUESTED** [review]

**Comment by fatedier** [comment]
By the way, you don't need to create a new PR, just push the new changes to the original branch.

File: pkg/config/v1/proxy.go [hunk 2]
```diff
@@ -97,6 +97,12 @@ type HealthCheckConfig struct {
 	// Path specifies the path to send health checks to if the
 	// health check type is "http".
 	Path string `json:"path,omitempty"`
+	// Headers specifies the headers to send health checks to if the
+	// health check type is "http".
+	Headers []HTTPHeader `json:"headers,omitempty"`
```
Comment (fatedier) [inline pkg/config/v1/proxy.go:102]:
```suggestion
	HTTPHeaders []HTTPHeader `json:"httpHeaders,omitempty"`
```

**Comment by KaiveYoung** [comment]
> By the way, you don't need to create a new PR, just push the new changes to the original branch.

New to PR, don’t know how to use it

**Review by fatedier — APPROVED** [review]

**Pull request merged**